packages/go_router/lib/src/delegate.dart

```diff
@@ -66,18 +106,37 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       pageKey: pageKey,
     );
 
-    _matches.push(newPageKeyMatch);
+    _matchList.push(newPageKeyMatch);
     notifyListeners();
   }
 
-  /// Returns `true` if there is more than 1 page on the stack.
+  /// Returns `true` if the active Navigator can pop.
   bool canPop() {
-    return _matches.canPop();
+    // Loop through navigators in reverse and call canPop()
+    final int matchCount = _matchList.matches.length;
+    for (int i = matchCount - 1; i >= 0; i -= 1) {
+      final RouteMatch match = _matchList.matches[i];
+      final RouteBase route = match.route;
+      if (route is GoRoute && route.parentNavigatorKey != null) {
+        final bool canPop = route.parentNavigatorKey!.currentState!.canPop();
```

Comment:
I would probably just return true, and do
assert(route.parentNavigatorKey!.currentState!.canPop())

Reply:
Why? Won't that mean that `canPop` is only called during development and not when the app is run in release mode?

Reply:
adc610d1ccd749b6f70cbaad81983f5ad4160a2f